### PR TITLE
assets: Ensure BPFFS mount (bsc#1146991)

### PR DESCRIPTION
**Pull request by mrostecki**
Cilium DaemonSet has the host mount from /sys/fs/bpf directory and it
expects a mount of BPFFS type. Otherwise, it creates its own BPFFS
mount, but it's not persistent. To avoid any issues with lack of the
mount on the host, ensure it in kubelet service

Signed-off-by: Michal Rostecki <mrostecki@suse.de>

### CODE REVIEW

File: internal/pkg/skuba/deployments/ssh/assets/assets.go [hunk 1]
```diff
@@ -40,6 +41,6 @@ EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
 # This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
 # the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
 EnvironmentFile=-/etc/sysconfig/kubelet
-ExecStart=
+ExecStartPre=/bin/bash -c "findmnt -t bpf --mountpoint /sys/fs/bpf || mount bpffs /sys/fs/bpf -t bpf"
```
Comment (evrardjp) [inline internal/pkg/skuba/deployments/ssh/assets/assets.go:44]: Shouldn't we keep the `ExecStart=` and only add the `ExecStartPre`?

Comment (evrardjp) [reply internal/pkg/skuba/deployments/ssh/assets/assets.go:44]: Also, the CI seems very unhappy right now :)

Comment (mrostecki) [reply internal/pkg/skuba/deployments/ssh/assets/assets.go:44]: What's the point of having two `ExecStart=` lines in one unit? Only the later one, `ExecStart=/usr/bin/kubelet ...` should stay IMO, as it's the relevant one.

Comment (mrostecki) [reply internal/pkg/skuba/deployments/ssh/assets/assets.go:44]: Done

File: internal/pkg/skuba/deployments/ssh/assets/assets.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ Description=kubelet: The Kubernetes Node Agent
 Documentation=http://kubernetes.io/docs/
 
 [Service]
+ExecStartPre=/bin/bash -c "findmnt -t bpf --mountpoint /sys/fs/bpf || mount bpffs /sys/fs/bpf -t bpf"
```
Comment (dirkmueller) [inline internal/pkg/skuba/deployments/ssh/assets/assets.go:26]: NIT: this prints a message, we might want to do this as

```
/bin/bash -c "findmnt -t bpf --mountpoint /sys/fs/bpf > /dev/null || mount bpffs /sys/fs/bpf -t bpf"
```

**Comment by dirkmueller** [comment]
So it looks like this is a missing feature in systemd. Starting with systemd 238, the bpffs is mounted by default. 

did we ask the systemd maintainers whether they'd be interested in backporting the fixes? that would fix all CaaSP releases at once.

**Comment by mrostecki** [comment]
> did we ask the systemd maintainers whether they'd be interested in backporting the fixes? that would fix all CaaSP releases at once.

We didn't.

**Comment by dirkmueller** [comment]
I asked now about a backport for https://github.com/systemd/systemd/commit/43b7f24b5e0dd048452112bfb344739764c58694

which would fix all caasp releases at the same time without having to get the caasp team to release updates on all caasp versions (which will take weeks to months)

**Comment by evrardjp** [comment]
I thought we were not allowed to backport systemd changes! That's amazing!

**Comment by dirkmueller** [comment]
@evrardjp Its not clear whether we can do the systemd backport (systemd is maintained by the systemd maintainers at SUSE). they have any right to reject such a backport or delay it until the next SP. 

however we need *a* solution in the caasp product nontheless. if it won't be done by systemd then there needs to be a caasp specific solution. it doesn't hurt to continue with this change though, a systemd fix will cause the code to become obsolete but it will not hurt anyone. once the systemd update is released (and sufficient time has passed to assume that customers have updated) the patch could simply be reverted.

**Comment by evrardjp** [comment]
tl;dr: I agree. (In fact I *agreed* else I wouldn't have merged).

I am not sure this change will be easily backportable into older branches for fixing caasp4.
It doesn't hurt to merge it for CaaSP5 until we figure it out with systemd.

I encourage the network team to think about the fact this won't fix CaaSP4, as this PR's target is master.
